Assets/Scripts/SS3D/Systems/Tile/TileSystem.cs

```diff
@@ -27,64 +33,67 @@ protected override void OnStart()
         }
 
         [ServerOrClient]
-        private IEnumerator WaitForResourcesLoad()
+        private async UniTask WaitForResourcesLoad()
         {
-            while (!Loader.IsInitialized)
-            {
-                yield return null;
-            }
+	        await UniTask.WaitUntil(() => Loader.IsInitialized);
 
             Load();
         }
 
-        [ServerOrClient]
-        private void Setup()
+        [Server]
+        private async void Setup()
         {
-            Loader = GetComponent<TileResourceLoader>();
-
-            // Server only loads the map
-            if (IsServer)
-            {
-                CreateMap("Test map");
-                StartCoroutine(WaitForResourcesLoad());
-                Punpun.Information(this, "All tiles loaded successfully");
-            }
+	        // Server only loads the map
+	        if (!IsServer)
+	        {
+		        return;
+	        }
+
+	        Loader = GetComponent<TileResourceLoader>();
+
+	        CreateMap(UnnamedMapName);
+
+	        await WaitForResourcesLoad();
+
+            Log.Information(this, "All tiles loaded successfully");
         }
 
         [ServerOrClient]
         private void CreateMap(string mapName)
         {
-            if (_currentMap == null)
-            {
-                TileMap map = TileMap.Create(mapName);
-                map.transform.SetParent(transform);
-                _currentMap = map;
-            }
+	        if (_currentMap != null)
```

Comment:
shoud we do this check with no warning ?

Reply:
Added a warning